lib/Backend/Commands.hs

```diff
@@ -237,10 +233,24 @@ findCmd (FindOptions pathMb textMb sortMb filters filterFields) = do
       OpLTE -> (<=)
       OpEQ -> (==)
 
-renameCmd :: (Member BackendEffect r, Member (Embed IO) r) => RenameOptions -> Sem r RenameResult
-renameCmd (RenameOptions oldPath newPath force) = do
-  -- TODO: reuse copy+delete operations
-  undefined
+renameCmd 
+  :: ( Member BackendEffect r
+     , Member (Embed IO) r
+     , Member (Error CofferError) r
+     , Member (Error RenameResult) r
+     ) 
+  => RenameOptions -> Sem r RenameResult
+renameCmd (RenameOptions oldPath newPath force) =
+  copyCmd (CopyOptions oldPath newPath force) >>= \case
+     cSuccess@(CPRSuccess _) -> 
+       if oldPath == newPath then
+         -- if oldPath equals to newPath then, we don't want to accidentally delete renamed directory/entry.
```

Comment:
Good catch!

Reply:
In fact, the `copyCmd` command should error out when `oldPath == newPath`. That's what the `cp` command does, and it seems like a sensible approach.

```
$ touch a
$ cp a a
cp: 'a' and 'a' are the same file
```

Could you please create a new issue for  this?

Reply:
`coffer copy` throws an error if two paths are equal. But if we add `-f` then there would be no errors. Is it expected behavior?

Reply:
> Is it expected behavior

It is the "expected behavior" in the sense that the [specification](https://www.notion.so/serokell/Password-sharing-2-232733503bb4469d85713d08f3b3a5e4#2459058a49df42ed84bcc37dd1d6fd4c) doesn't say anything about this particular scenario (when the 2 paths are the same). So in that sense, `coffer copy` behaving the exact same way for 2 different paths as it does for 2 equal paths is the expected behavior.

But what I'm proposing is that we consider this edge case, and consider changing `coffer copy`'s behavior. I think it should always throw an error when the 2 paths are the same, like the `cp` command, regardless of whether `-f` was used. 

We should create an issue to discuss this and see if the rest of the team agrees.